src/aind_metadata_service/tars/mapping.py

```diff
@@ -0,0 +1,148 @@
+            prep_type = VirusPrepType.CRUDE
+            prep_protocol = PrepProtocols.SOP_VC002.value
+        elif viral_prep_type == ViralPrepTypes.PURIFIED_SOP.value:
+            prep_type = VirusPrepType.PURIFIED
+            prep_protocol = PrepProtocols.SOP_VC003.value
+        elif viral_prep_type == ViralPrepTypes.CRUDE_HGT.value:
+            prep_type = VirusPrepType.CRUDE
+            prep_protocol = PrepProtocols.HGT.value
+        elif viral_prep_type == ViralPrepTypes.RABIES_SOP.value:
+            prep_type = None
+            prep_protocol = PrepProtocols.SOP_VC001.value
+        elif viral_prep_type == ViralPrepTypes.CRUDE_PHP_SOP.value:
+            prep_type = VirusPrepType.CRUDE
+            prep_protocol = PrepProtocols.SOP_VC004.value
+        elif viral_prep_type == ViralPrepTypes.CRUDE_MGT2.value:
+            prep_type = VirusPrepType.CRUDE
+            prep_protocol = PrepProtocols.MGT2.value
+        elif viral_prep_type == ViralPrepTypes.CRUDE_MGT1.value:
+            prep_type = VirusPrepType.CRUDE
+            prep_protocol = PrepProtocols.MGT1.value
+        elif viral_prep_type == ViralPrepTypes.PURIFIED_MGT1.value:
+            prep_type = VirusPrepType.PURIFIED
+            prep_protocol = PrepProtocols.MGT1.value
+        elif viral_prep_type == ViralPrepTypes.CRUDE_HGT1.value:
+            prep_type = VirusPrepType.CRUDE
+            prep_protocol = PrepProtocols.HGT1.value
+        else:
+            # TODO: figure out how we want to handle the rest
+            #  PHP_SOP_UW, VTC_AAV1, UNKNOWN, and IODIXANOL
+            prep_type = None
+            prep_protocol = None
+        return prep_type, prep_protocol
+
+    @staticmethod
+    def _convert_datetime(date: str):
+        """Converts date string to datetime date"""
+        # Is it safe to assume it'll always be in this pattern
+        date_pattern = re.compile(r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$")
+
+        if not date_pattern.match(date):
+            print(
+                "Invalid date format."
+                " Please provide a date in the format: YYYY-MM-DDTHH:MM:SSZ"
+            )
+            return None
+
+        return datetime.strptime(date, "%Y-%m-%dT%H:%M:%SZ")
+
+    @staticmethod
+    def _map_virus_aliases(aliases: list) -> tuple[str, str, str]:
+        """Maps aliases to full_genome_name, material_id, viral prep id"""
+        material_id, viral_prep_id, full_genome_name = None, None, None
+        for alias in aliases:
+            name = alias["name"]
+            if VirusAliasPatterns.AIP.value.match(name):
+                material_id = name
+            elif VirusAliasPatterns.AIV.value.match(name):
+                viral_prep_id = name
+            else:
+                # TODO: check pattern
+                full_genome_name = name
+        return material_id, viral_prep_id, full_genome_name
+
+    def map_response_to_injection_materials(
+        self, response
+    ) -> InjectionMaterial:
```

Comment:
There's not guarantee a viable model can be constructed. Should make it an Optional and handle cases where the pydantic model fails validation